Homework/2/Program.cs

```diff
@@ -49,62 +64,112 @@ static void Main(string[] args)
                 Console.WriteLine("11 - Log out\n");
             }
             else {
-                Console.WriteLine("Select option:\n");
+                Console.WriteLine("Select option:\n\n");
                 Console.WriteLine("1 - Log in\n");
                 Console.WriteLine("2 - Register\n");
             }
 
+            Console.WriteLine("12 - Close console\n");
+
             OptionSelection(option: Console.ReadLine());
         }
 
+        // Loading users from file to array - HW3  / User details should follow this format: username;password - in different line for each user.
+        // Things to consider: line validation - multiple ;, empty usernames/passwords // ; char should also be validated on user creation
+        static void LoadUsersFromFile() {
+            try
+            {
+                using (StreamReader reader = new StreamReader(pathToFile))
+                {
+                    while (!reader.EndOfStream)
+                    {
+                        string[] userCredentials = reader.ReadLine().Split(';');
+                        if (CheckIfUserExists(userCredentials[0]))
+                        {
+                            throw new DuplicateUserCredentialsException(userCredentials[0]);
+                        }
+                        else
+                        {
+                            AddUserToArrayAtIndex(userCredentials, 0);
+                        }
+                    }
+                }
```

Comment:
Could move this into a function of its own, will be more clear what are you actually trying.

Reply:
I thought since UsersNotFoundExecption was called if no users.txt file was found, that it would be ok if I used FileNotFoundExecption instead.

Added UsersNotFoundException exeption an moved reader to it's own function.